org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/console/JavaStackTraceHyperlink.java

```diff
@@ -188,11 +192,56 @@ public IStatus runInUIThread(IProgressMonitor monitor) {
 						JDIDebugUIPlugin.statusDialog(ConsoleMessages.JavaStackTraceHyperlink_3, status);
 					}
 				} else if (source instanceof List) { // ambiguous
+
 					@SuppressWarnings("unchecked")
 					List<Object> matches = (List<Object>) source;
 					int line = lineNumber + 1; // lineNumber starts with 0, but line with 1, see #linkActivated
+					if (generatedLink == null) { // Handles invalid links (without line number)
+						OpenFromClipboardAction.handleMatches(matches, line, typeName, ConsoleMessages.JavaDebugStackTraceHyperlink_dialog_title);
+						return Status.OK_STATUS;
+					}
+					List<Object> exactMatchesFiltered = new ArrayList<>();
+					int methodNameStartIndex = generatedLink.indexOf('.');
+					int methodNameEndIndex = generatedLink.lastIndexOf(')');
+					if (methodNameStartIndex != -1 && methodNameEndIndex != -1) {
+						String methodSignature = generatedLink.substring(methodNameStartIndex + 1, methodNameEndIndex + 1).replaceAll(" ", ""); //$NON-NLS-1$//$NON-NLS-2$
+						String methodNameExtracted = methodSignature.substring(0, methodSignature.indexOf('('));
+						for (Object obj : matches) {
+							if (obj instanceof IType type) {
+								try {
+									if (extractFromResults(type, methodSignature, methodNameExtracted)) {
+										exactMatchesFiltered.add(obj);
+									} else if (generatedLink.indexOf('$') != -1) { // checks for inner class
```

Comment:
Can we please extract `if (generatedLink.indexOf('$') != -1)` block into a dedicated method?

Reply:
Yes